File: java/eyes.selenium.java/src/main/java/com/applitools/eyes/selenium/ShadowDomLocator.java
package com.applitools.eyes.selenium;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

/**
 * shadow dom locator
 */
public class ShadowDomLocator extends TargetPathLocator {

  public ShadowDomLocator(TargetPathLocator parent, PathNodeValue value) {
    super(parent, value);
  }

  public RegionLocator region(WebElement element) {
    return new RegionLocator(this, new ElementReference(element));
  }

  public RegionLocator region(By by) {
    return new RegionLocator(this, new ElementSelector(by));
  }

  public RegionLocator region(String selector) {
    return new RegionLocator(this, new ElementSelector(selector)); // this
  }

  public ShadowDomLocator shadow(WebElement element) {
    return new ShadowDomLocator(this, new ElementReference(element));
  }

  public ShadowDomLocator shadow(By by) {
    return new ShadowDomLocator(this, new ElementSelector(by));
  }

  public ShadowDomLocator shadow(String selector) {
    return new ShadowDomLocator(this, new ElementSelector(selector)); //this
  }
}

File: dotnet/Eyes.Selenium.DotNet/Fluent/ShadowDomLocator.cs
using OpenQA.Selenium;

namespace Applitools.Selenium
{
    public class ShadowDomLocator : TargetPathLocator
    {
        public ShadowDomLocator(TargetPathLocator parent, IPathNodeValue value)
            : base(parent, value)
        {
        }

        public RegionLocator Region(IWebElement element)
        {
            return new RegionLocator(this, new ElementReference(element));
        }

        public RegionLocator Region(By by)
        {
            return new RegionLocator(this, new ElementSelector(by));
        }

        public RegionLocator Region(string selector)
        {
            return new RegionLocator(this, new ElementSelector(selector)); // this
        }

        public ShadowDomLocator Shadow(IWebElement element)
        {
            return new ShadowDomLocator(this, new ElementReference(element));
        }

        public ShadowDomLocator Shadow(By by)
        {
            return new ShadowDomLocator(this, new ElementSelector(by));
        }

        public ShadowDomLocator Shadow(string selector)
        {
            return new ShadowDomLocator(this, new ElementSelector(selector)); //this
        }
    }
}